### PR TITLE
Feature/cryto delete

### PR DESCRIPTION
**Pull request by janaakhterov**
Add cryto_delete

### CODE REVIEW

File: src/transaction_crypto_delete.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    pub fn delete_account_id(&mut self, id: AccountId) {
+        self.inner().delete_account_id = Some(id);
+    }
+}
+
+impl ToProto<TransactionBody_oneof_data> for TransactionCryptoDelete {
+    fn to_proto(&self) -> Result<TransactionBody_oneof_data, Error> {
+        let mut data = proto::CryptoDelete::CryptoDeleteTransactionBody::new();
+
+        match self.transfer_account_id {
```
Comment (mehcode) [inline src/transaction_crypto_delete.rs:43]: Default this to the operator

File: src/transaction_crypto_delete.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    delete_account_id: Option<AccountId>,
+}
+
+interfaces!(
+    TransactionCryptoDelete: Any,
+    ToProto<TransactionBody_oneof_data>
+);
+
+impl Transaction<TransactionCryptoDelete> {
+    pub fn crypto_delete(client: &Client) -> Self {
```
Comment (mehcode) [inline src/transaction_crypto_delete.rs:20]: Require the "account ID to delete"

File: src/transaction_crypto_delete.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        Self::new(
+            client,
+            TransactionCryptoDelete {
+                transfer_account_id: None,
+                delete_account_id: None,
+            },
+        )
+    }
+
+    pub fn transfer_account_id(&mut self, id: AccountId) {
```
Comment (mehcode) [inline src/transaction_crypto_delete.rs:30]: `.transfer_to()` perhaps? Not 100% on the name

File: src/transaction_crypto_delete.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        self.inner().transfer_account_id = Some(id);
+    }
+
+    pub fn delete_account_id(&mut self, id: AccountId) {
+        self.inner().delete_account_id = Some(id);
```
Comment (mehcode) [inline src/transaction_crypto_delete.rs:35]: Each builder method needs to return `&mut Self`

File: src/bridge/key.rs [hunk 5]
```diff
@@ -117,16 +121,26 @@ pub unsafe extern "C" fn hedera_public_key_from_str(s: *const c_char, out: *mut
...
+    message_len: usize,
+) -> i8 {
     debug_assert!(!s.is_null());
     debug_assert!(!message.is_null());
     debug_assert!(!p.is_null());
-    debug_assert!(!out.is_null());
 
     let message = slice::from_raw_parts(message, message_len);
 
-    *out = try_ffi!((*p).verify(message, &(*s))) as i8;
```
Comment (mehcode) [inline src/bridge/key.rs]: This looks like a merge conflict gone odd. This file has regressed to an earlier commit.

**Review by mehcode — CHANGES_REQUESTED** [review]

File: src/transaction_crypto_delete.rs (hunk 1 above)
Comment (janaakhterov) [reply src/transaction_crypto_delete.rs:43]: Can you elaborate?

**Pull request merged**